### PR TITLE
Added Ascension Revision

### PR DESCRIPTION
**Pull request by StellarWitch7**
Adds a new revision pattern to bring an inner circle *up*, replacing its parent.

### CODE REVIEW

**Comment by StellarWitch7** [comment]
Fixed Ascension Revision not working on the glyph of the root node. Also made glyphs persist after using non-destructive revisions on the same grid.

**Comment by StellarWitch7** [comment]
Added Collection Stratagem, a glyph that combines all input lists into one.

**Comment by StellarWitch7** [comment]
> Fixed Ascension Revision not working on the glyph of the root node. ~Also made glyphs persist after using non-destructive revisions on the same grid.~ (Appears to work practically all the time, but I saw _one_ exception that I've not been able to reproduce.)

I was able to reproduce the issue and resolve it.

**Comment by StellarWitch7** [comment]
Here are a couple changes I've made I'm not sure you want, so lemme know if you want me to revert either of them. 

`src/client/java/dev/enjarai/trickster/ModKeyBindings.java`: added boolean check for players who don't want to have to take their top hat out of their offhand to change hotbar selection. 

Made the Absorption Distortion glyph support taking a list as the first input, where it will then treat the first three elements of the list as the normal input and construct a new vector from them.

File: src/client/java/dev/enjarai/trickster/screen/SpellPartWidget.java [hunk 1]
```diff
@@ -274,34 +276,56 @@ protected void stopDrawing() {
...
             }
+        } else if (compiled.equals(EXPAND_TO_OUTER_CIRCLE_GLYPH)) {
+            if (drawingPart != spellPart) {
+                boolean b = spellPart.glyph == drawingPart;
+                setSubPartInTree(drawingPart, Optional.of(drawingPart), spellPart, true);
+
+                if (b) {
+                    drawingPart = spellPart;
+                }
+            }
```
Comment (enjarai) [inline src/client/java/dev/enjarai/trickster/screen/SpellPartWidget.java:287]: If `drawingPart == spellPart`, you can skip the recursive search entirely to save some processing.

File: src/main/java/dev/enjarai/trickster/spell/SpellPart.java [hunk 2]
```diff
@@ -31,16 +31,17 @@ public final class SpellPart implements Fragment {
     public static final Endec<SpellPart> ENDEC = CodecUtils.toEndec(CODEC);
 
     public Fragment glyph;
+    public Fragment oldGlyph;
```
Comment (enjarai) [inline src/main/java/dev/enjarai/trickster/spell/SpellPart.java:34]: Explained in discord.

**Review by enjarai — APPROVED** [review]
Should be good now.

**Pull request merged**